File: arrays/121. maximum-profit/maximum_profit.py
import sys


class Solution:
    """Leetcode https://leetcode.com/problems/best-time-to-buy-and-sell-stock/"""

    def maxProfit(self, prices: List[int]) -> int:
        max_sum = 0
        running_profit = 0
        min_ele = sys.maxsize
        for element in prices:
            running_profit = element - min_ele
            max_sum = max(running_profit, max_sum)
            if min_ele > element:
                min_ele = element

        return max_sum

File: arrays/121. maximum-profit/maximum_profit.rb
def max_profit(prices)
 len = prices.length
 max_diff = 0
 for i in 0 .. len - 1
   max_ele = prices[i..len].max
   ele_diff = max_ele - prices[i]
   if ele_diff > max_diff
     max_diff = ele_diff
   end
 end
 max_diff
end

## max_profit([7,1,5,3,6,4])